src/main/java/stringaddcalculator/StringAddCalculator.java

```diff
@@ -0,0 +1,64 @@
+package stringaddcalculator;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String REGEX_FORMAT = "[%s]";
+    private static final String DEFAULT_DELIMITERS = ",:";
+    private static final String CUSTOM_FORMAT = "//(.)\n(.*)";
+
+    public static int splitAndSum(String input) {
+
+        if (isEmptyString(input)) {
+            return 0;
+        }
+
+        return operate(input);
+    }
+
+    private static int operate(String input) {
+        int result = 0;
+
+        String[] operands = splitInput(input);
+        for (String operand : operands) {
+            result += getValidValue(operand);
+        }
+
+        return result;
+    }
+
+    private static int getValidValue(String input) {
+        int value = parseValidInt(input);
+
+        if (value < 0) {
+            throw new RuntimeException();
+        }
+
+        return value;
+    }
+
+    private static int parseValidInt(String input) {
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException();
+        }
```

Comment:
이 부분은 몇가지 생각해볼만한 부분이 있어서 Comment 남겨요 (이미 고려하고 작성하셨을 수도 있지만요 ㅎ;)
결론적으로는 저도 위와 유사하게 작성할 것 같습니다.

```suggestion
    private int parse(String value) {
        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException e) {
            log.debug(e.getMessage());
            return DEFAULT_VALUE; // 이 부분은 '빈 문자열 또는 null 값을 입력할 경우 0을 반환해야 한다.(예 : “” => 0, null => 0)' 요구사항에 해당됩니다만, 저는 동의하지 않는 요구사항입니다.
        }
    }
```

우선, try-catch문으로 분기 흐름을 제어하는 것은 좋지 않은 구성이라고 생각합니다. 
그리고 현재 Exception을 흘려보내고 있는 것 역시 안티 패턴으로 알고 있어요

>https://www.slipp.net/questions/350

---

아래의 코드는 Optional을 사용하므로 비용효율적이지 않다고 생각할 수 있습니다.

```java
   private int parse(String value) {
        return Optional.ofNullable(value)
                .map(Integer::parseInt)
                .orElse(DEFAULT_VALUE);
    }
```

---

그렇다고 명시적으로 null check 등을 외부로 드러내는 것은, 
1. parseInt 내부에서 null 체크를 하는데 중복으로 확인하는 것이 불필요하지 않은가
2. null, 빈값 등의 예외처리를 비즈니스 로직에 드러냄으로써 얻는 이점은 무엇인가
등의 관점에서 불필요하다고 생각합니다.

Reply:
요구사항 테스트에 있는 RuntimeException을 맞춰야 한다는 생각에 갇혀있었는데 지금 다시보니 parseInt에서 나오는 NumberFormatException이 결국 runtimeException을 상속받게 되는군요ㅠㅠ
말씀주신 대로 try-catch로 흐름을 제어하는 부분과 exception으로 흘려보내는 패턴들도 개선할 수 있을 것 같습니다!
피드백 감사합니다:)